flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-mysql/src/main/java/org/apache/flink/cdc/connectors/mysql/source/reader/MySqlPipelineRecordEmitter.java

```diff
@@ -90,7 +94,15 @@ public MySqlPipelineRecordEmitter(
     protected void processElement(
             SourceRecord element, SourceOutput<Event> output, MySqlSplitState splitState)
             throws Exception {
-        if (isLowWatermarkEvent(element) && splitState.isSnapshotSplitState()) {
+        if (shouldEmitAllCtesInSnapshotMode
+                && StartupOptions.snapshot().equals(sourceConfig.getStartupOptions())) {
```

Comment:
We can calculate `Boolean isBounded = StartupOptions.snapshot().equals(sourceConfig.getStartupOptions()` in constructor instead of calculate it per element.